multichain-testing/test/xcs-swap-anything/helpers.ts

```diff
@@ -0,0 +1,826 @@
+
+      destinationClient = client;
+      destinationAddress = address;
+      destinationWallet = wallet;
+    } else {
+      destinationChain = 'osmosis';
+      destinationClient = osmosisClient;
+      destinationAddress = osmosisAddress;
+      destinationWallet = osmosisWallet;
+    }
+
+    const destinationBalanceBefore =
+      await destinationClient.getAllBalances(destinationAddress);
+    t.log(
+      'destination wallet balances before transfer: ',
+      destinationBalanceBefore,
+    );
+
+    const transferArgs = makeIBCTransferMsg(
+      { denom: issuingDenom, value: amount },
+      { address: destinationAddress, chainName: destinationChain },
+      { address: issuingAddress, chainName: issuingChain },
+      Date.now(),
+      useChain,
+    );
+
+    // TODO #9200 `sendIbcTokens` does not support `memo`
+    // @ts-expect-error spread argument for concise code
+    const txRes = await issuingClient.sendIbcTokens(...transferArgs);
+    if (txRes && txRes.code !== 0) {
+      console.error(txRes);
+      throw Error(`failed to ibc transfer funds to ${destinationAddress}`);
+    }
+    t.is(txRes.code, 0, `Transaction succeeded`);
+    console.log('sendIbcTokens TxResponse: ', txRes);
+
+    const denom = await getDenomHash(
+      destinationChain,
+      issuingChain,
+      issuingDenom,
+    );
+
+    const getDenomAmount = balances => {
+      const balance = balances.find(coin => coin.denom === `ibc/${denom}`);
+      const amount = balance ? Number(balance.amount) : 0;
+      return amount;
+    };
+
+    await retryUntilCondition(
+      () => destinationClient.getAllBalances(destinationAddress),
+      (coins: Coin[]) => {
+        const before = getDenomAmount(destinationBalanceBefore);
+        const current = getDenomAmount(coins);
+        return current > before;
+      },
+      `Transferred tokens not found on destination address: ${destinationAddress}`,
+    );
+    console.log(
+      `Transferred tokens found on destination address: ${destinationAddress}`,
+    );
+
+    return {
+      client: destinationClient,
+      address: destinationAddress,
+      wallet: destinationWallet,
+    };
+  };
+
+  const invokeOsmosisContract = async (contractAddress: string, txMsg) => {
+    const clientRegistry = osmosisClient.registry;
+    clientRegistry.register(MsgExecuteContract.typeUrl, MsgExecuteContract);
+
+    const fee = {
+      amount: [{ denom: 'uosmo', amount: '1000' }],
+      gas: '200000',
+    };
+
+    const message = MsgExecuteContract.fromPartial({
+      sender: osmosisAddress,
+      contract: contractAddress,
+      msg: toUtf8(JSON.stringify(txMsg)),
+      funds: [],
+    });
+
+    const encodeObjects: Array<import('@cosmjs/proto-signing').EncodeObject> = [
+      {
+        typeUrl: '/cosmwasm.wasm.v1.MsgExecuteContract',
+        value: message,
+      },
+    ];
+
+    const response = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      encodeObjects,
+      fee,
+    );
```

Comment:
I think it's more clear with `encodeObjects` inlined:

```suggestion
    const response = await osmosisClient.signAndBroadcast(
      osmosisAddress,
      [{ typeUrl: MsgExecuteContract.typeUrl, value: message }],
      fee,
    );
```

Reply:
Changes introduced in commit: e487bb1d7487aa1964323a6bf95cb83e8641e847